tests/devices/qubit/test_apply_operation.py

```diff
@@ -824,11 +848,38 @@ def test_double_excitation(self, method):
 
         assert qml.math.allclose(state_v1, state_v2)
 
+    @pytest.mark.parametrize("apply_wires", ([0, 3], [0, 1, 3, 2], [2, 1], [1, 3]))
+    def test_grover(self, method, apply_wires):
+        """Tests a four qubit GroverOperator."""
+        op = qml.GroverOperator(apply_wires)
+        new_state = method(op, self.state)
+
+        expected_state = np.copy(self.state)
+        for _op in op.decomposition():
+            expected_state = method(_op, expected_state)
+
+        assert qml.math.allclose(expected_state, new_state)
+
+
+@pytest.mark.parametrize(
+    "num_wires, einsum_called, tensordot_called",
+    [(2, True, False), (3, False, True), (9, False, False)],
+)
+def test_grover_dispatching(num_wires, einsum_called, tensordot_called, mocker):
+    """Test that apply_grover dispatches to einsum correctly for small numbers of wires."""
+    op = qml.GroverOperator(list(range(num_wires)))
+    state = np.zeros([2] * num_wires, dtype=complex)
+    spy_einsum = mocker.spy(qml.math, "einsum")
+    spy_tensordot = mocker.spy(qml.math, "argsort")
```

Comment:
Cannot use `math.tensordot` to find out whether `apply_operation_tensordot` was called, because the kernel also uses `math.tensordot`.
`math.argsort` is used by `apply_operation_tensordot` and not by `apply_operation_einsum` or the kernel, so I used that for now.
Is there a proper way to spy on `apply_operation_tensordot` directly?

Reply:
For testing, I'd mostly just be concerned about "is it correct in all three regimes?".   

For "is it performant in all three regimes?": that's a separate question. We could potentially think about adding a `pytest-benchmark` to this, but it might be overkill at this point in time.

Reply:
Right :) So is the used placeholder `spy` okay to check whether `apply_operation_tensordot` was called?

Reply:
I think it's fine. we do something similar to test the `apply_operation` dispatching for ParametrizedEvolution. If the code ever changes, tests will fail and we'll fix 'em!